src/copy.jl

```diff
@@ -137,3 +137,15 @@ function Base.copy(model::AbstractModel)
     new_model, _ = copy_model(model)
     return new_model
 end
+
+"""
+    deepcopy(_::AbstractModel)
+
+Calls error. Calling `deepcopy` over a JuMP model is not supported, nor
+planned to be supported, because it would involve making a deep copy of
+the underlying solver (behind a C pointer). Check [`copy`](@ref) instead.
+"""
+function Base.deepcopy(_::AbstractModel)
```

Comment:
The issue only applies for `Model`, not necessarily `AbstractModel`.

Reply:
This approach forces people sub-typing `AbstractModel` to explicitly opt-in to `deepcopy` which I think is a good thing.

Reply:
Yes, but `deepcopy` is opt-out in Julia, so not sure if we should override that just for `JuMP.AbstractModel`.

Reply:
Unless we want to create a style guideline about `deepcopy`.

Reply:
So, what will be? I have changed to Model, but I may change back to AbstractModel. I had made the change before seeing @odow's comment. I think  both sides have some points, and no decision will be final anyway, as new subclasses of AbstractModel may define specializations of `deepcopy`. This is basically my first contribution, so I think the decision should not be mine.

Reply:
Leave it as `Model`. You contribution is appreciated! Sorry for the back-and-forward.

Reply:
I think it can be merged then. I have left it as Model, and changed the parameter style and removed the trailing line, as asked by others. I did not intend to sound rude, XD, it is just I was worried the merge was waiting because you people were expecting my input.